### ISSUE TITLE
About the use of Deep Convolutional Generative Adversarial Network for synthetic medical data

### ISSUE THREAD

**Issue by bsauty** [ivadomed/ivadomed#16]
Here are a few things I found related to the use of the generative models to obtain synthetic medical data.

## Bibliography

-  **Synthesis in Multi-Contrast MRI with Conditional Generative Adversarial Networks** (February 2019) : https://ieeexplore.ieee.org/abstract/document/8653423
:arrow_forward:  reports on the use of an adversarial loss function and a GAN architecture for synthesizing one contrast from another one using conditional GANs, outperforms other "non linear interpolation mapping between contrasts". Very high results but can only generate one contrast from one other with already existing database (for instance predict t2w or FLAIR from t1w to save time). Works better than all existing [end-to-end NN](https://arxiv.org/abs/1801.06940) for domain adaptation. 

-  **Synthetic Medical Images from Dual Generative Adversarial Networks** (January 2018) : https://arxiv.org/abs/1709.01872
:arrow_forward:  general idea of dual generative networks and demonstration of liability on retina segmentation with a U-net

-  **Deep Generative Adversarial Neural Networks for Realistic Prostate Lesion MRI Synthesis** (August 2017) : https://arxiv.org/abs/1708.00129
:arrow_forward:  results less enthusiastic but is a good explanation of GANs and how it could benefit medical databases with a precisely defined working architecture.

-  **Medical Image Synthesis for Data Augmentation and Anonymization Using Generative Adversarial Networks** (Septembre 2018): https://link.springer.com/chapter/10.1007/978-3-030-00536-8_1
:arrow_forward:  brain tumor MRI generation lead to improved segmentation + similar results when training on synthetic or on real data 

-  **Synthetic data augmentation using GAN for improved liver lesion classification** (April 2018) https://ieeexplore.ieee.org/abstract/document/8363576 and https://www.sciencedirect.com/science/article/pii/S0925231218310749
:arrow_forward:  massive increase in efficiency for a classification task on CT scans

-  **Medical Image Synthesis with Context-Aware Generative Adversarial Networks** (September 2017) : https://link.springer.com/chapter/10.1007/978-3-319-66179-7_48
:arrow_forward:  predicting CT from MRI using an adversarial strategy and outperforms by far all existing strategies (end-to-end NN, contexted random forests and atlas based methods). Useful to avoid the radiations for the patient and help for PET scan attenuation. Uses an [Auto-Context Model](http://pages.ucsd.edu/~ztu/publication/cvpr08_autocontext.pdf) to obtain better results and seems very promising to help GANs converge so it is worth investigating. :warning: difficulty to obtain the necessary database.

## General idea

1. Data augmentation through the generation of large amounts of synthetic data can work for certain tasks, and there are reports of better segmentation through U-nets after generating new synthetic data.
2. Conditional GANs have already proven to be able to generate new contrasts from existing data for MRI. ⚠️ until now no physical equation has been taken into account so only know contrasts can be generated.
3. There are many reports in different fields (CT scans/MRI acquisition and segmentation/classification tasks) reporting increased accuracy and selectivity with the use of GANs.
4. All the articles are quite recent.

### Possible architectures to investigate 
- [Conditional GAN](https://github.com/eriklindernoren/PyTorch-GAN#conditional-gan)
▶️ Condition both the generator and discriminator with one or many labels. Generator takes noise (+ the label obviously) as input and this could be useful for database expansion.
- [CycleGAN](https://github.com/eriklindernoren/PyTorch-GAN#cyclegan)
▶️ Learning mapping from one domain to another one with one generator for both ways. Could be useful to deduce one contrast from another one. This particular case can also work with unpaired images (ie. translate an image from one domain to another one without having access to the target result but only a collection of images from the target domain)
- [Semi-Supervised GAN](https://github.com/eriklindernoren/PyTorch-GAN#semi-supervised-gan)
▶️ Add a classifier network and train the trio (G/D/C) together to generate more accurate image of each domain. Could be used for database expansion.

### Possible functionalities to implement
- Known contrast generation from only one quick acquisition (t2w or flair from t1w) to save time in the IRM.
- Expand our database for each known contrast by generating new data with a classic convolutional GAN.
- Deduce the CT scan from the MR acquisition (would require a new database of coupled CT/MR data but could be an interesting functionality for sct). It is already being done for the brain MRI .
- Implement the physical NMR equations within a context aware GAN (as suggested in #11) to be able to generate any contrast on a continuous spectrum and obtain a robust database for the FiLM architecture to train on. (Never been done before in the literature so maybe a little too optimistic but worth a try I guess 🤓)

## Other approaches 

- Multimodal MR Synthesis via Modality-Invariant Latent Representation
 : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5904017/
- Random Forest Regression for Magnetic Resonance Image Synthesis : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5099106/

This issue is related to #7 but I figured opening one to discuss this particular way to generate new contrast coud make sense and be less messy.
